_inc/client/plans/single-product-backup.jsx

```diff
@@ -158,6 +159,25 @@ class SingleProductBackupBody extends React.Component {
 		this.props.setSelectedBackupType( event.target.value );
 	};
 
+	handleUpgradeButtonClick = selectedBackupType => () => {
+		const page = startCase( window.location.hash.replace( /#\//g, ' ' ).trim() );
```

Comment:
That requires a few changes to bring it down to this component, but I wonder if we could use the `route` prop that is available a few levels higher? Then we wouldn't need to rely on `startCase`, and we would get a path that matches the slug, and thus other tracks events in the dashboard like `jetpack_wpa_page_view`; I think that may make it easier to build funnels.

Something like this?

```diff
diff --git a/_inc/client/main.jsx b/_inc/client/main.jsx
index 0edbc2977..165975414 100644
--- a/_inc/client/main.jsx
+++ b/_inc/client/main.jsx
@@ -210,6 +210,7 @@ class Main extends React.Component {
 						siteRawUrl={ this.props.siteRawUrl }
 						siteAdminUrl={ this.props.siteAdminUrl }
 						rewindStatus={ this.props.rewindStatus }
+						route={ this.props.route }
 					/>
 				);
 				break;
diff --git a/_inc/client/plans/index.jsx b/_inc/client/plans/index.jsx
index 129926b7b..48fea6d60 100644
--- a/_inc/client/plans/index.jsx
+++ b/_inc/client/plans/index.jsx
@@ -18,7 +18,7 @@ import { getSitePlan } from 'state/site';
 
 export class Plans extends React.Component {
 	render() {
-		const { dailyBackupUpgradeUrl, products, realtimeBackupUpgradeUrl } = this.props;
+		const { dailyBackupUpgradeUrl, products, route, realtimeBackupUpgradeUrl } = this.props;
 
 		const upgradeLinks = {
 			daily: dailyBackupUpgradeUrl,
@@ -30,7 +30,11 @@ export class Plans extends React.Component {
 				<QueryProducts />
 				<QuerySite />
 				{ products && 'jetpack_free' === get( this.props.sitePlan, 'product_slug' ) && (
-					<SingleProductBackup products={ products } upgradeLinks={ upgradeLinks } />
+					<SingleProductBackup
+						products={ products }
+						route={ route }
+						upgradeLinks={ upgradeLinks }
+					/>
 				) }
 				<PlanGrid />
 			</React.Fragment>
diff --git a/_inc/client/plans/single-product-backup.jsx b/_inc/client/plans/single-product-backup.jsx
index 5617d742c..6a800d018 100644
--- a/_inc/client/plans/single-product-backup.jsx
+++ b/_inc/client/plans/single-product-backup.jsx
@@ -4,7 +4,7 @@
 import React, { Fragment, useState } from 'react';
 import PropTypes from 'prop-types';
 import { translate as __ } from 'i18n-calypso';
-import { get, startCase } from 'lodash';
+import { get } from 'lodash';
 
 /**
  * Internal dependencies
@@ -16,66 +16,74 @@ import PlanPrice from 'components/plans/plan-price';
 
 import './single-product-backup.scss';
 
-export function SingleProductBackup( { products, upgradeLinks } ) {
-	const [ selectedBackupType, setSelectedBackupType ] = useState( 'real-time' );
-
-	const billingTimeFrame = 'yearly';
-	const currencyCode = get( products, [ 'jetpack_backup_daily', 'currency_code' ], '' );
-
-	const priceDaily = get( products, [ 'jetpack_backup_daily', 'cost' ], '' );
-	const priceDailyMonthly = get( products, [ 'jetpack_backup_daily_monthly', 'cost' ], '' );
-	const priceDailyMonthlyPerYear = '' === priceDailyMonthly ? '' : priceDailyMonthly * 12;
-	const priceRealtime = get( products, [ 'jetpack_backup_realtime', 'cost' ], '' );
-	const priceRealtimeMonthly = get( products, [ 'jetpack_backup_realtime_monthly', 'cost' ], '' );
-	const priceRealtimeMonthlyPerYear = '' === priceRealtimeMonthly ? '' : priceRealtimeMonthly * 12;
-
-	const backupOptions = [
-		{
-			type: 'daily',
-			name: __( 'Daily Backups' ),
-			discountedPrice: priceDaily,
-			fullPrice: priceDailyMonthlyPerYear,
-		},
-		{
-			type: 'real-time',
-			name: __( 'Real-Time Backups' ),
-			discountedPrice: priceRealtime,
-			fullPrice: priceRealtimeMonthlyPerYear,
-		},
-	];
+export function SingleProductBackup( { products, route, upgradeLinks } ) {
+					const [ selectedBackupType, setSelectedBackupType ] = useState( 'real-time' );
 
-	return (
-		<React.Fragment>
-			<h1 className="plans-section__header">{ __( 'Solutions' ) }</h1>
-			<h2 className="plans-section__subheader">
-				{ __( "Just looking for backups? We've got you covered." ) }
-			</h2>
-			<div className="plans-section__single-product">
-				<div className="single-product-backup__accented-card dops-card">
-					<div className="single-product-backup__accented-card-header">
-						<h3 className="single-product-backup__header-title">{ __( 'Jetpack Backup' ) }</h3>
-						<SingleProductBackupPriceGroup
-							billingTimeFrame={ billingTimeFrame }
-							currencyCode={ currencyCode }
-							discountedPrice={ [ priceDaily, priceRealtime ] }
-							fullPrice={ [ priceDailyMonthlyPerYear, priceRealtimeMonthlyPerYear ] }
-						/>
-					</div>
-					<div className="single-product-backup__accented-card-body">
-						<SingleProductBackupBody
-							billingTimeFrame={ billingTimeFrame }
-							currencyCode={ currencyCode }
-							backupOptions={ backupOptions }
-							selectedBackupType={ selectedBackupType }
-							setSelectedBackupType={ setSelectedBackupType }
-							upgradeLinks={ upgradeLinks }
-						/>
-					</div>
-				</div>
-			</div>
-		</React.Fragment>
-	);
-}
+					const billingTimeFrame = 'yearly';
+					const currencyCode = get( products, [ 'jetpack_backup_daily', 'currency_code' ], '' );
+
+					const priceDaily = get( products, [ 'jetpack_backup_daily', 'cost' ], '' );
+					const priceDailyMonthly = get( products, [ 'jetpack_backup_daily_monthly', 'cost' ], '' );
+					const priceDailyMonthlyPerYear = '' === priceDailyMonthly ? '' : priceDailyMonthly * 12;
+					const priceRealtime = get( products, [ 'jetpack_backup_realtime', 'cost' ], '' );
+					const priceRealtimeMonthly = get(
+						products,
+						[ 'jetpack_backup_realtime_monthly', 'cost' ],
+						''
+					);
+					const priceRealtimeMonthlyPerYear =
+						'' === priceRealtimeMonthly ? '' : priceRealtimeMonthly * 12;
+
+					const backupOptions = [
+						{
+							type: 'daily',
+							name: __( 'Daily Backups' ),
+							discountedPrice: priceDaily,
+							fullPrice: priceDailyMonthlyPerYear,
+						},
+						{
+							type: 'real-time',
+							name: __( 'Real-Time Backups' ),
+							discountedPrice: priceRealtime,
+							fullPrice: priceRealtimeMonthlyPerYear,
+						},
+					];
+
+					return (
+						<React.Fragment>
+							<h1 className="plans-section__header">{ __( 'Solutions' ) }</h1>
+							<h2 className="plans-section__subheader">
+								{ __( "Just looking for backups? We've got you covered." ) }
+							</h2>
+							<div className="plans-section__single-product">
+								<div className="single-product-backup__accented-card dops-card">
+									<div className="single-product-backup__accented-card-header">
+										<h3 className="single-product-backup__header-title">
+											{ __( 'Jetpack Backup' ) }
+										</h3>
+										<SingleProductBackupPriceGroup
+											billingTimeFrame={ billingTimeFrame }
+											currencyCode={ currencyCode }
+											discountedPrice={ [ priceDaily, priceRealtime ] }
+											fullPrice={ [ priceDailyMonthlyPerYear, priceRealtimeMonthlyPerYear ] }
+										/>
+									</div>
+									<div className="single-product-backup__accented-card-body">
+										<SingleProductBackupBody
+											billingTimeFrame={ billingTimeFrame }
+											currencyCode={ currencyCode }
+											backupOptions={ backupOptions }
+											route={ route }
+											selectedBackupType={ selectedBackupType }
+											setSelectedBackupType={ setSelectedBackupType }
+											upgradeLinks={ upgradeLinks }
+										/>
+									</div>
+								</div>
+							</div>
+						</React.Fragment>
+					);
+				}
 
 function SingleProductBackupPriceGroup( {
 	billingTimeFrame,
@@ -150,6 +158,7 @@ class SingleProductBackupBody extends React.Component {
 		backupOptions: PropTypes.array,
 		billingTimeFrame: PropTypes.string,
 		currencyCode: PropTypes.string,
+		route: PropTypes.object,
 		setSelectedBackupType: PropTypes.func,
 		selectedBackupType: PropTypes.string,
 		upgradeLinks: PropTypes.object,
@@ -159,14 +168,13 @@ class SingleProductBackupBody extends React.Component {
 		this.props.setSelectedBackupType( event.target.value );
 	};
 
-	handleUpgradeButtonClick = selectedBackupType => () => {
-		const page = startCase( window.location.hash.replace( /#\//g, ' ' ).trim() );
+	handleUpgradeButtonClick = ( selectedBackupType, route ) => () => {
 
 		analytics.tracks.recordJetpackClick( {
 			target: `upgrade-${ selectedBackupType }`,
 			type: 'upgrade',
 			product: selectedBackupType,
-			page,
+			page: route.path,
 		} );
 	};
 
@@ -183,6 +191,7 @@ class SingleProductBackupBody extends React.Component {
 			backupOptions,
 			billingTimeFrame,
 			currencyCode,
+			route,
 			selectedBackupType,
 			upgradeLinks,
 		} = this.props;
@@ -234,7 +243,7 @@ class SingleProductBackupBody extends React.Component {
 						<div className="single-product-backup__upgrade-button-container">
 							<Button
 								href={ upgradeLinks[ selectedBackupType ] }
-								onClick={ this.handleUpgradeButtonClick( selectedBackupType ) }
+								onClick={ this.handleUpgradeButtonClick( selectedBackupType, route ) }
 								primary
 							>
 								{ upgradeTitles[ selectedBackupType ] }

```

Reply:
Good idea @jeherve but I don't think we need to pass it down. We can totally use the `withRouter` HoC to make it simpler. I'll work on that. Thanks for the suggestion!

Reply:
@jeherve I've suggested using the existing router in https://github.com/Automattic/jetpack/pull/14103/commits/1aa0afdae486fbd2d9f6bdc4d1c07321ed2441d6 - could you take another look?